extensions/common/aws/aws-s3-core/src/main/java/org/eclipse/edc/aws/s3/AwsClientProviderImpl.java

```diff
@@ -168,4 +170,8 @@ private void handleEndpointOverride(SdkClientBuilder<?, ?> builder) {
             builder.endpointOverride(endpointOverride);
         }
     }
+
+    private void handlePathStyleEnabled(S3BaseClientBuilder<?, ?> builder) {
```

Comment:
why not having a `createS3Configuration` method that returns the `S3Configuration` instance and then it can be set on every builder? it would avoid the side effect

Reply:
It makes sense. I refactored the code based on your suggestion.